pkg/kubectl/cmd/run.go

```diff
@@ -106,6 +107,23 @@ func Run(f *cmdutil.Factory, cmdIn io.Reader, cmdOut, cmdErr io.Writer, cmd *cob
 		return cmdutil.UsageError(cmd, "NAME is required for run")
 	}
 
+	// Let kubectl run follow rules for `--`
+	var commandAndCustomArgs []string
```

Comment:
This commit doesn't fix these cases, it fixes #13004 . And run command hasn't -p operation.